src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -450,20 +482,56 @@ def _prepare_profiler(self, config: SolverConfig, run_cfg) -> None:
             self._profiler = None
 
     def _get_results(self, config: SolverConfig):
-        best_cost, idx = self.solver_process.optimum.get()
-        best_cost = SolutionReadoutPyModel.decode_cost(best_cost)
-        best_state = self._get_best_state(config, idx)
-        best_timestep = self.solver_process.solution_step.aliased_var.get() - 2
-        return best_state, int(best_cost), int(best_timestep)
+        idx = 0
+        if self.solver_process.is_discrete:
+            best_cost, idx = self.solver_process.optimum.get()
+            best_cost = SolutionReadoutPyModel.decode_cost(best_cost)
+            best_timestep = (
+                self.solver_process.solution_step.aliased_var.get() - 2
+            )
+        best_state = self._get_best_state(config, int(idx))
+
+        if self.solver_process.is_discrete:
+            return best_state, int(best_cost), int(best_timestep)
+        else:
+            return best_state, None, None

```

Comment:
Why do continuous problems not return the best cost and best time step?

Reply:
because the QP solver doesn't use the Solution Reader and read gate yet